usaspending_api/agency/v2/views/object_class_list.py

```diff
@@ -0,0 +1,66 @@
+from django.db.models import Q, Sum, F
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any, List
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, ListMixin
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_simple_pagination_metadata
+from usaspending_api.references.models import ObjectClass
+
+
+class ObjectClassList(ListMixin, AgencyBase):
+    """
+    Obtain the list of object classes for a specific agency in a single
+    fiscal year based on whether or not that object class has ever
+    been submitted in File B.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/object_class.md"
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        results = list(self.get_object_class_list())
+        count = len(results)
+        results = results[self.pagination.lower_limit : self.pagination.upper_limit]
+        page_metadata = get_simple_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        page_metadata["count"] = count
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "limit": self.pagination.limit,
+                "page_metadata": page_metadata,
+                "results": results[: self.pagination.limit],
+                "messages": self.standard_response_messages,
+            }
+        )
+
+    def get_object_class_list(self) -> List[dict]:
+        q = (
+            Q(financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe__gt=0)
+            | Q(financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe__lt=0)
+            | Q(financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe__gt=0)
+            | Q(financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe__lt=0)
+        )
+        filters = {
```

Comment:
This is slightly nitpicky, but why have `q` and `filter` separated? It seems cleaner to me to move the filters to Q objects and just have one array.

Reply:
No good reason and will clean that up.